agentops/agent.py

```diff
@@ -0,0 +1,139 @@
+from .event import Event
+from .http import HttpClient
+from .helpers import safe_serialize, get_ISO_time, Providers
+from openai import Stream, AsyncStream
+from openai.types.chat import ChatCompletionChunk
+from openai.resources import AsyncCompletions
+import openai
+
+
+class Agent:
+    def __init__(self, config, client, name: str, session_id: str, provider: Providers = Providers.OPEN_AI):
+        self.config = config
+        self.name = name
+        self._session_id = session_id
+        self.client = client
+        self.event_stream = None
+
+        payload = {
+            "name": name,
+            "session_id": session_id,
+            "provider": provider
+        }
+
+        serialized_payload = \
+            safe_serialize(payload).encode("utf-8")
+        res = HttpClient.post(f'{self.config.endpoint}/agents',
+                              serialized_payload,
+                              self.config.api_key)
+
+        # TODO: when we have internal logs, track any non-200 response here
+        self.id = res.body.get('id')
+
+        self.chat_completion_original = None
+        self.async_chat_completion_original = None
+        if provider == Providers.OPEN_AI:
+            self.chat_completion_original = openai.chat.completions.create
+        else:
+            raise Exception('Unknown LLM provider {}'.format(provider))
+
+    def chat_completion(self, *args, **kwargs):
+        init_timestamp = get_ISO_time()
+
+        """Handle responses for OpenAI versions >v1.0.0"""
+
+        response = self.chat_completion_original(*args, **kwargs)
+        # response = Completions.create(*args, **kwargs)
+
+        # if the response is a generator, decorate the generator
+        if isinstance(response, Stream):
+            def generator():
+                for chunk in response:
+                    self._handle_stream_chunk(chunk, init_timestamp, **kwargs)
+                    yield chunk
+
+            return generator()
```

Comment:
This is copied from `llm_tracker.py`, no? Why not update that instrumentor instead?